### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by lastquestion**
This is largely adding more tests.

We move to nock ~2.x so we can actually reject with error. There is more cleanup needed for tests, but trying to keep PR's small again. This diff has a lot of changes, but it's because I've also taken a little time to try and __coffeescriptize__ the code (removing unnecessary parens and the like).

### CODE REVIEW

File: test/unit/DiscoveryAnnouncerSpec.coffee [hunk 1]
```diff
@@ -51,6 +51,29 @@ describe "DiscoveryAnnouncer", ->
...
+            expect(announcementRequest.serviceType).to.equal "test"
+            expect(announcementRequest.serviceUri).to.equal "test"
+            expect(announcementRequest.announcementId).to.be.ok
+            [201, announcementRequest]
+
+      @announcer.announce
+        serviceType: "test"
+        serviceUri: "test"
+      .then (result) ->
+        expect(result).to.have.property('announcementId').to.be.ok
```
Comment (virtue3) [inline test/unit/DiscoveryAnnouncerSpec.coffee]: you can use my reg-ex for a guid for this from the constructor tests you still haven't merged :)

File: test/unit/DiscoveryLongPollerSpec.coffee [hunk 2]
```diff
@@ -41,6 +41,12 @@ describe "DiscoveryLongPoller", ->
       @discoveryLongPoller.schedulePoll = done
       @discoveryLongPoller.startPolling()
 
+    it "starting polling twice has no effect", ->
+      @discoveryLongPoller.schedulePoll = sinon.spy()
+      @discoveryLongPoller.startPolling()
+      @discoveryLongPoller.startPolling()
+      expect(@discoveryLongPoller.schedulePoll.callCount).to.equal 1
```
Comment (virtue3) [inline test/unit/DiscoveryLongPollerSpec.coffee]:
```coffeescript
process.nextTick () -> 
```
before the expect(callcount) check?

File: test/unit/DiscoveryAnnouncerSpec.coffee (hunk 1 above)
Comment (lastquestion) [inline test/unit/DiscoveryAnnouncerSpec.coffee]: your PR doesn't merge.

Comment (virtue3) [inline test/unit/DiscoveryAnnouncerSpec.coffee]: yea... cuz you merged yours first :dancer:

File: test/unit/DiscoveryLongPollerSpec.coffee (hunk 2 above)
Comment (lastquestion) [inline test/unit/DiscoveryLongPollerSpec.coffee]: doesn't need it

File: test/unit/DiscoveryAnnouncerSpec.coffee [hunk 3]
```diff
@@ -51,6 +51,30 @@ describe "DiscoveryAnnouncer", ->
...
+            expect(announcementRequest.serviceUri).to.equal "test"
+            expect(announcementRequest.announcementId).to.be.ok
+            [201, announcementRequest]
+
+      @announcer.announce
+        serviceType: "test"
+        serviceUri: "test"
+      .then (result) ->
+        expect(result).to.have.property('announcementId')
+          .to.match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/i)
```
Comment (virtue3) [inline test/unit/DiscoveryAnnouncerSpec.coffee]: @lastquestion I just added it for you . it was in poi-microsite. sorry.  

If this is gtg then I'll merge the PR.

**Pull request merged**